plugins/module_utils/network/sonic/config/vlan_mapping/vlan_mapping.py

```diff
@@ -0,0 +1,494 @@
+        """
+        result = {'changed': False}
+        warnings = list()
+
+        existing_vlan_mapping_facts = self.get_vlan_mapping_facts()
+        commands, requests = self.set_config(existing_vlan_mapping_facts)
+        if commands:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.code)
+            result['changed'] = True
+        result['commands'] = commands
+
+        changed_vlan_mapping_facts = self.get_vlan_mapping_facts()
+
+        result['before'] = existing_vlan_mapping_facts
+        if result['changed']:
+            result['after'] = changed_vlan_mapping_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_vlan_mapping_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = remove_empties_from_list(self._module.params['config'])
+        have = existing_vlan_mapping_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        state = self._module.params['state']
+        have = self.convert_vlan_ids_range(have)
+        want = self.convert_vlan_ids_range(want)
+        diff = get_diff(want, have, TEST_KEYS)
+
+        if state == 'overridden':
+            commands, requests = self._state_overridden(want, have, diff)
+        elif state == 'deleted':
+            commands, requests = self._state_deleted(want, have)
+        elif state == 'merged':
+            commands, requests = self._state_merged(want, have, diff)
+        elif state == 'replaced':
+            commands, requests = self._state_replaced(want, have, diff)
+
+        ret_commands = remove_empties_from_list(commands)
+        return ret_commands, requests
+
+    def _state_replaced(self, want, have, diff):
+        """ The command generator when state is replaced
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        requests = []
+        commands = []
+        commands_del = []
+
+        if diff:
+            commands_del = self.get_replaced_delete_list(want, have)
```

Comment:
When a mapping M is in both want and have, the attributes in want.M have same values as those attribues in have.M, but want.M has less attributes than have.M, so diff=[]. The result should be that have.M is replaced by want.M? Please verify.

Reply:
After doing some testing I see what you mean the "if diff:" is causing the commands_del to not run when it should so I have removed it and checking the changes now.

Reply:
I have pushed the fix for this. I had to make some more changes for errors that occurred after removing the diff check.